nimbleclient/v1/restclient.py

```diff
@@ -167,21 +171,42 @@ def close_connection(self):
     def get(self, endpoint, **params):
         """Wrapper for GET requests"""
 
-        url=f'https://{self.hostname}:{self.port}/{endpoint}'
+        try:
+            return self.get_data(endpoint, **params)
+
+        except requests.exceptions.RequestException as error:
+            logging.exception(error)
+            raise ConnectionError(f"Error retrieving data from {self.hostname}")
+
+    def get_data(self, endpoint, filter=None, **params):
```

Comment:
Why is this in a separate function? Just make the base `get()` function perform the action while taking afvanced filters into consideration.

Reply:
Agreed, will merge into single function